src/Microsoft.VisualStudio.Composition/Reflection/TypeRef.cs

```diff
@@ -242,7 +257,26 @@ public TypeRef MakeGenericTypeRef(ImmutableArray<TypeRef> genericTypeArguments)
 
             // We use the resolver parameter instead of the field here because this TypeRef instance
             // might have been constructed by TypeRef.Get(Type) and thus not have a resolver.
-            return new TypeRef(this.Resolver, this.AssemblyName, this.assemblyId, this.MetadataToken, this.FullName, this.TypeFlags, this.GenericTypeParameterCount, genericTypeArguments);
+            return new TypeRef(this.Resolver, this.AssemblyName, this.assemblyId, this.MetadataToken, this.FullName, this.TypeFlags, this.GenericTypeParameterCount, genericTypeArguments, this.ElementTypeRef);
+        }
+
+        public bool IsAssignableFrom(Type other)
```

Comment:
Is the goal to only avoid an assembly load on the *receiving* side? If you want to avoid loading *either* assembly, should you perhaps take `TypeRef` instead of `Type` for your parameter?

Reply:
The minimal goal was to only avoid the assembly load on the receiving side. From what I understood the general case for improvement would be a modified MEF cache, in that case won't most exporting types have to have their assembly loaded already to scan for new parts?

Reply:
Keep in mind that the assignability checks happen *during composition*, which applies to *all* assemblies, whether they were changed or not. So if you load the 'right side' of this test, then a *lot* of assemblies will load during composition because most assemblies export *something*, whether or not they were changed in the last cache invalidating update.

Reply:
Ah, I thought the checks only happened on the boundary between modified and non-modified assemblies. In that case what I think will be necessary is recording the type hierarchy inside the typerefs.